apps/admin/backend/src/app.ts

```diff
@@ -166,13 +171,81 @@ function buildApi({
       return auth.unprogramCard(constructAuthMachineState(workspace));
     },
 
-    async writeBallotPackageSignatureFile(input: {
-      ballotPackagePath: string;
-    }): Promise<void> {
-      await artifactAuthenticator.writeSignatureFile({
-        type: 'ballot_package',
-        path: input.ballotPackagePath,
+    async saveBallotPackageToUsb(): Promise<Result<void, 'no_usb_drive'>> {
+      await logger.log(LogEventId.SaveBallotPackageInit, 'election_manager');
+
+      const ballotPackageZipStream = new ZipStream();
+      const electionDefinition = getCurrentElectionDefinition(workspace);
+      assert(electionDefinition !== undefined);
+      const systemSettings =
+        store.getSystemSettings() ?? DEFAULT_SYSTEM_SETTINGS;
+      await addFileToZipStream(ballotPackageZipStream, {
+        path: 'election.json',
+        contents: electionDefinition.electionData,
+      });
+      await addFileToZipStream(ballotPackageZipStream, {
+        path: 'systemSettings.json',
+        contents: JSON.stringify(systemSettings, null, 2),
       });
+      ballotPackageZipStream.finalize();
+
+      const tempDirectory = dirSync().name;
+      const ballotPackageFileName = generateFilenameForBallotExportPackage(
+        electionDefinition,
+        new Date()
+      );
+      await fs.writeFile(
+        join(tempDirectory, ballotPackageFileName),
+        ballotPackageZipStream
+      );
+
+      const usbMountPoint = (await usb.getUsbDrives())[0]?.mountPoint;
+      if (!usbMountPoint) {
+        await logger.log(
+          LogEventId.SaveBallotPackageComplete,
+          'election_manager',
+          {
+            disposition: 'failure',
+            message: 'Error saving ballot package: no USB drive',
+            result: 'Ballot package not saved, error shown to user.',
+          }
+        );
+        return err('no_usb_drive');
```

Comment:
In this circumstance the temporary directory will never be cleaned up. Put the cleanup in a `finally` block?

Reply:
Nice catch! https://github.com/votingworks/vxsuite/pull/3554/commits/4c6ff323bb93c4e5131127670406aec797bb7e3f?w=1